bootstrap/bin/hmc.ml

```diff
@@ -0,0 +1,42 @@
+open Basis
+include Basis.Rudiments
+open Hmc
+
+let scan_file path =
+  let open Format in
+  let rec fn scanner = begin
+    let scanner', ctoken = Scan.next scanner in
+    let atoken = Scan.ConcreteToken.atoken ctoken in
+    let source = Scan.ConcreteToken.source ctoken in
+    printf "  %a : %s\n"
+      Scan.Source.pp_loc source
+      (Scan.AbstractToken.to_string atoken)
+    ;
+    match atoken with
+    | Scan.AbstractToken.Tok_end_of_input -> ()
+    | _ -> fn scanner'
+  end in
+  printf "@[<h>";
+  let () = match File.of_path ~flag:File.Flag.R_O path with
```

Comment:
It looks like I set the default as `RW` and that it combines `O_RDRW | O_CREAT`. I kinda wonder if the default should be `R_O`.

Reply:
Read-write by default seems okay, but creating by default is really surprising. I think we should change the default to either `R_O` or `RW_O`. Which do you think is the better choice? I have a slight preference for `R_O`, because it is the least permissive, and commonly what is wanted anyway.